spring-boot/admin/src/test/java/eu/coatrack/admin/service/GatewayHealthMonitorServiceTest.java

```diff
@@ -0,0 +1,92 @@
+package eu.coatrack.admin.service;
+
+import eu.coatrack.admin.model.repository.ProxyRepository;
+import eu.coatrack.api.Proxy;
+import eu.coatrack.api.ProxyStates;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.mockito.Mockito.when;
+
+public class GatewayHealthMonitorServiceTest {
+
+    @Mock
+    ProxyRepository proxyRepository;
+
+    @Mock
+    Proxy proxySample;
+
+    private List<Proxy> sampleList = new ArrayList<>();
+
+    @InjectMocks
+    private GatewayHealthMonitorService gatewayHealthMonitorService;
+
+    public Proxy createSampleProxy(boolean monitoredEnabled, String proxyName) {
+        Proxy proxy = new Proxy();
+        proxy.setName(proxyName);
+        proxy.setId("GatewayIDTest");
+        proxy.setMonitoringEnabled(monitoredEnabled);
+        proxy.updateTimeOfLastSuccessfulCallToAdmin_setToNow();
+        return proxy;
+    }
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        sampleList = Arrays.asList(createSampleProxy(true, "test1"), createSampleProxy(false, "test2"), proxySample);
```

Comment:
"test1" and "test2" are very generic names, at least use names like "testProxy1", even better would be to use names that are meaningful for the test e.g. "testProxyMonitoringEnabled", "testProxyMonitoringDisabled"

Reply:
Done